ArchiSteamFarm/CardsFarmer.cs

```diff
@@ -347,16 +376,36 @@ internal sealed class CardsFarmer : IDisposable {
 
 				float hours = 0;
 
-				Match match = Regex.Match(hoursString, @"[0-9\.,]+");
-				if (match.Success) {
-					if (!float.TryParse(match.Value, NumberStyles.Number, CultureInfo.InvariantCulture, out hours)) {
+				Match hoursMatch = Regex.Match(hoursString, @"[0-9\.,]+");
+				if (hoursMatch.Success) {
+					if (!float.TryParse(hoursMatch.Value, NumberStyles.Number, CultureInfo.InvariantCulture, out hours)) {
 						Logging.LogNullError(nameof(hours), Bot.BotName);
 						return;
 					}
 				}
 
-				GamesToFarm[appID] = hours;
+				games.Add(new Game {
+					AppID = appID,
+					HoursPlayed = hours,
+					CardsRemaining = cardsRemaining
+				});
+			}
+
+			IEnumerable<Game> gamesToFarm;
+			switch (Bot.BotConfig.FarmingOrder) {
+				case BotConfig.EFarmingOrder.MostCardDropRemainingFirst:
+					gamesToFarm = games.OrderByDescending(g => g.CardsRemaining);
+					break;
+
+				case BotConfig.EFarmingOrder.FewestCardDropRemainingFirst:
+					gamesToFarm = games.OrderBy(g => g.CardsRemaining);
+					break;
+
+				default:
+					gamesToFarm = games;
+					break;
 			}
+			GamesToFarm.AddRange(gamesToFarm);
```

Comment:
List<T> is not thread-safe and ```CheckPage()``` is executed by many concurrent threads at the same time. Consider using ```ConcurrentHashSet<Game>```.